Network device broken in emulator on linux

### Before submitting

- [x] Reproduced the issue with external plugins and user-patches disabled.
- [x] Searched existing issues for similar reports.
- [x] Tested on the latest nightly build ([more information](/koreader/koreader/issues/11068)).

### Device

Other (please specify in additional information)

### KOReader version

2026.03-110

### Model and firmware version

emulator on arch linux

### Issue description

Working on updates for a plugin, I noticed that now if you enable wifi in the emulator, it crashes with the below stacktrace. This used to work, and I know there's a couple of solutions, but before wasting your time with a patch, is this a known/intended issue? I didn't see any existing issues, and `git blame` suggested there hadn't been any direct updates lately (other than the change in sdl versions maybe?), but figured I'd ask first. Thanks!

```
05/16/26-14:15:44 DEBUG _refresh: Enqueued ui update for region 0 0 810 629 dithering: true 
05/16/26-14:15:44 DEBUG refresh on logical rectangle 810x629+0+0 
05/16/26-14:15:46 DEBUG NetworkListener: onNetworkConnected 
05/16/26-14:15:46 DEBUG NetworkListener: unschedule network activity check 
05/16/26-14:15:46 DEBUG NetworkListener: network activity check 
./luajit: frontend/ui/network/networklistener.lua:103: attempt to concatenate a nil value
stack traceback:
	frontend/ui/network/networklistener.lua:103: in function '_getTxPackets'
	frontend/ui/network/networklistener.lua:137: in function '_scheduleActivityCheck'
	frontend/ui/network/networklistener.lua:215: in function 'handleEvent'
	frontend/ui/widget/container/widgetcontainer.lua:83: in function 'propagateEvent'
	frontend/ui/widget/container/widgetcontainer.lua:101: in function 'handleEvent'
	frontend/ui/uimanager.lua:973: in function 'broadcastEvent'
	frontend/device/sdl/device.lua:403: in function 'action'
	frontend/ui/uimanager.lua:1015: in function '_checkTasks'
	frontend/ui/uimanager.lua:1461: in function 'handleInput'
	frontend/ui/uimanager.lua:1581: in function 'run'
	reader.lua:301: in main chunk
	[C]: at 0x55e2920ef83b
make: *** [make/emulator.mk:20: run] Error 1
```

### Regression

Yes

### Last working version

_No response_

### Steps to reproduce

1. Compile emulator. Maybe linux speciffic somehow? (or worse, arch specific?)
2. Open menu to enable networking
3. Enable wireless
4. Give it a few seconds to try and cat the stats on the net device


### Log

```shell

```

### Additional information

_No response_

--- a/frontend/ui/network/networklistener.lua
+++ b/frontend/ui/network/networklistener.lua
@@ -181,7 +181,8 @@ function NetworkListener:onNetworkConnected()
     NetworkMgr:setWifiState(true)
     NetworkMgr:setConnectionState(true)
 
-    if not G_reader_settings:isTrue("auto_disable_wifi") then
+    -- You can't set auto_disable_wifi on devices without getNetworkInterfaceName() but guard against it in case it was accidentally set anyway.
+    if not self:getNetworkInterfaceName() or not G_reader_settings:isTrue("auto_disable_wifi") then
         return
     end
 

